include/zephyr/sys/util.h

```diff
@@ -421,6 +421,21 @@ extern "C" {
  */
 #define IN_RANGE(val, min, max) ((val) >= (min) && (val) <= (max))
 
+/**
+ * @brief Map a value from one scale to another.
+ *
+ * A value of an input range is mappedto a value of an output range.
+ * The ranges are described by their min and max values.
+ *
+ * @param i The value to convert from the input range to the output range.
+ * @param imin The minimum value of the input range.
+ * @param imax The maximum value of the input range.
+ * @param omin The minimum value of the output range.
+ * @param omax The maximum value of the output range.
+ */
+#define MAP_VALUE(i, imin, imax, omin, omax) \
```

Comment:
MAP may be a non optimal fit here, since maps map any input to any output, possibly in both directions, and even mapping 1 to many values :)
```
0 -> 100
3 -> 67
5 -> 45
7 -> (3, 5)
...
```
while scaling linearly maps an input to an output, you even hint at this in the brief :)
```
#define SCALE_VALUE(...)
```
would be better fitting IMO.

Reply:
I am tempted to use `SCALE()`: as out of the 5 pages of [search results](https://github.com/search?q=%22%23define+SCALE%28%22+language%3AC&type=code&l=C&p=5
), it shew-up 5 times with that same meaning ([1](https://github.com/phoboslab/high_impact/blob/cbeda1483947de5c4bf699d30509137a3583a590/src/utils.h#L50) [2](https://github.com/ElectricRCAircraftGuy/eRCaGuy_hello_world/blob/800da6c2b65aa6afd74bb6da1250de3cbd55eaa8/c/utilities.h#L154) [3](https://github.com/Alhadis/Research-Unix-v10/blob/59b9cdae24725c9832d4ff615cf22507d26796f3/v10/cmd/movie/libj.c#L634) [4](https://github.com/Tinkerforge/wifi-v2-extension/blob/2f9e3b67b5b13dc3aae7b063ebd07a5711dbf428/software/src/utility.h#L39) [5](https://github.com/Alhadis/Research-Unix-v8/blob/389623b76d5b6e195361f0705b1826b00ae14d19/jerq/src/lib/jj/mhit.c#L7))